### PR TITLE
Add stax support

### PR DESCRIPTION
**Pull request by xchapron-ledger**

### CODE REVIEW

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -135,4 +137,4 @@ jobs:
       - name: Run test
         env:
           CTEST_OUTPUT_ON_FAILURE: 1
-        run: pytest --tb=short -v tests/ --device ${{ matrix.model }}
+        run: PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python pytest --tb=short -v tests/ --device ${{ matrix.model }}
```
Comment (xchapron-ledger) [inline .github/workflows/ci-workflow.yml:140]: Should not be necessary once https://github.com/LedgerHQ/ledgerctl/pull/51 is merged and deployed

File: src/io.h [hunk 2]
```diff
@@ -17,42 +18,81 @@ void io_seproxyhal_display(const bagl_element_t *element);
...
+void io_start_interruption_timeout();
 
 /**
- * Receive APDU command in G_io_apdu_buffer.
- *
- * @return zero or positive integer if success, -1 otherwise.
- *
+ * Removes the timeout started from io_start_interruption_timeout.
+ */
+void io_clear_interruption_timeout();
```
Comment (agrojean-ledger) [inline src/io.h:38]: Related to Stax ?

File: tests/test_sign_cmd.py [hunk 3]
```diff
@@ -95,13 +111,26 @@ def test_sign_tx_refused(backend, navigator, test_name):
...
-                                                  "Reject",
-                                                  ROOT_SCREENSHOT_PATH,
-                                                  test_name)
-
-    assert client.get_async_response().status == Errors.SW_DENY
+    if firmware.device.startswith("nano"):
+        with client.sign_tx(path=path, transaction=transaction):
+            # Disable raising when trying to unpack an error APDU
+            backend.raise_policy = RaisePolicy.RAISE_NOTHING
+            navigator.navigate_until_text_and_compare(NavIns(NavInsID.RIGHT_CLICK),
```
Comment (agrojean-ledger) [inline tests/test_sign_cmd.py:118]: We can directly use `NavInsID` with the latest ragger version

File: Makefile [hunk 4]
```diff
@@ -22,7 +22,7 @@ endif
 include $(BOLOS_SDK)/Makefile.defines
 
 APP_LOAD_PARAMS  = --curve secp256k1
-ifeq ($(TARGET_NAME), TARGET_NANOX)
+ifeq ($(TARGET_NAME),$(filter $(TARGET_NAME),TARGET_NANOX TARGET_STAX))
```
Comment (sgliner-ledger) [inline Makefile:25]: From what I've understood, this is related to the BLE functonnality, should we use the `HAVE_BLE` instead ? 
Anyway, I think this should probably be moved to the SDK.

File: Makefile [hunk 5]
```diff
@@ -92,9 +102,13 @@ LDLIBS  += -lm -lgcc -lc
...
 APP_SOURCE_PATH += src
-SDK_SOURCE_PATH += lib_stusb lib_stusb_impl lib_ux
+SDK_SOURCE_PATH += lib_stusb lib_stusb_impl
+
+ifneq ($(TARGET_NAME),TARGET_STAX)
+SDK_SOURCE_PATH += lib_ux
+endif
 
-ifeq ($(TARGET_NAME),TARGET_NANOX)
+ifeq ($(TARGET_NAME),$(filter $(TARGET_NAME),TARGET_NANOX TARGET_STAX))
```
Comment (sgliner-ledger) [inline Makefile:111]: Same here, maybe we should use the HAVE_BLE flag ? :thinking:

File: Makefile [hunk 6]
```diff
@@ -49,27 +51,35 @@ DEFINES += APPNAME=\"$(APPNAME)\"
...
-DEFINES += HAVE_BAGL HAVE_UX_FLOW HAVE_SPRINTF HAVE_SNPRINTF_FORMAT_U
+DEFINES += HAVE_SPRINTF HAVE_SNPRINTF_FORMAT_U
 DEFINES += HAVE_IO_USB HAVE_L4_USBLIB IO_USB_MAX_ENDPOINTS=6 IO_HID_EP_LENGTH=64 HAVE_USB_APDU
 DEFINES += USB_SEGMENT_SIZE=64
 DEFINES += BLE_SEGMENT_SIZE=32
 DEFINES += HAVE_WEBUSB WEBUSB_URL_SIZE_B=0 WEBUSB_URL=""
 DEFINES += UNUSED\(x\)=\(void\)x
 
-ifeq ($(TARGET_NAME),TARGET_NANOX)
+ifeq ($(TARGET_NAME),$(filter $(TARGET_NAME),TARGET_NANOX TARGET_STAX))
```
Comment (sgliner-ledger) [inline Makefile:61]: same

File: src/ui/nbgl_display_transaction.c [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+// Buffer where the transaction amount string is written
+static char g_amount[30];
+// Buffer where the transaction address string is written
+static char g_address[43];
+
+static nbgl_layoutTagValue_t pairs[2];
+static nbgl_layoutTagValueList_t pairList;
+static nbgl_pageInfoLongPress_t infoLongPress;
+
+static void review_continue();
```
Comment (sgliner-ledger) [inline src/ui/nbgl_display_transaction.c:52]: You forgot the arg here :)

File: src/ui/menu_nbgl.c [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+#include "os.h"
+#include "glyphs.h"
+#include "nbgl_use_case.h"
+
+#include "../globals.h"
+#include "menu.h"
+
+/*********************
+ *      DEFINES
```
Comment (sgliner-ledger) [inline src/ui/menu_nbgl.c:33]: Not a big fan of these empty placeholders :)

**Review by sgliner-ledger — APPROVED** [review]

File: Makefile (hunk 4 above)
Comment (xchapron-ledger) [reply Makefile:25]: Keeping these for your PR adding a lot of stuff in SDK

**Pull request merged**